app/services/book_service.go

```diff
@@ -0,0 +1,26 @@
+package services
+
+import (
+	"github.com/ilie20088/go-web-app-boilerplate/app/models"
+	"github.com/ilie20088/go-web-app-boilerplate/app/repositories"
+)
+
+// BookFetcher is an interfaces that defines set of business operations related to books
+type BookFetcher interface {
+	FetchBook(id string) (*models.Book, error)
+}
+
+// BookFetcherImpl is an implementation of BookFetcher
+type BookFetcherImpl struct{}
+
+var bookRepository repositories.BookRepository
+
+// InitBookService initializes book service
+func InitBookService(_bookRepository repositories.BookRepository) {
+	bookRepository = _bookRepository
+}
+
+// FetchBook fetches book by id or returns a error returned by book repository
+func (b BookFetcherImpl) FetchBook(id string) (*models.Book, error) {
+	return bookRepository.FetchBookByID(id)
+}
```

Comment:
Don't forget that this code will be read for reference. In this state it is more confusing than useful in this case. I think it would be better to leave this kind of choices to the developer that will expand this instead of optimizing prematurely.